app/views/StatusView.js

```diff
@@ -88,14 +88,31 @@ class StatusView extends React.Component {
 
 	submit = async() => {
 		logEvent(events.STATUS_DONE);
-		const { statusText } = this.state;
+		const { statusText, status } = this.state;
 		const { user } = this.props;
 		if (statusText !== user.statusText) {
 			await this.setCustomStatus();
 		}
+		if (status !== user.status) {
+			await this.setUserStatus();
+		}
 		this.close();
 	}
 
+	setUserStatus = async() => {
+		const { status, statusText } = this.state;
+		logEvent(events[`STATUS_${ status.toUpperCase() }`]);
+		try {
+			const result = await RocketChat.setUserStatus(status, statusText);
+			if (result.success) {
+				store.dispatch(setUser({ status }));
+			}
+		} catch (e) {
+			logEvent(events.SET_STATUS_FAIL);
+			log(e);
+		}
+	}
+
```

Comment:
This code can be further simplified and DRY. Instead of creating a separate function, we can use this.setCustomStatus function for both(updating user status & user status text). But The exact error may not be given to the user.